### PR TITLE
Add some helper functions to File.h

### PR DESCRIPTION
**Pull request by dimitry-**
They work with std::filesystem::* counterparts in exception free way.

Test: run OrbitBaseTests

### CODE REVIEW

**Review by beckerhe — APPROVED** [review]
It looks good to me, but it seems Windows doesn't agree.

File: src/OrbitBase/FileTest.cpp [hunk 1]
```diff
@@ -267,4 +269,52 @@ TEST(File, ReadStructureAtOffset) {
...
+TEST(File, ListFilesInDirectory) {
+  auto tmp_file_or_error = TemporaryFile::Create();
+  ASSERT_THAT(tmp_file_or_error, HasNoError());
+  auto tmp_file = std::move(tmp_file_or_error.value());
+  auto file_list_or_error = ListFilesInDirectory(tmp_file.file_path().parent_path());
+  ASSERT_THAT(file_list_or_error, HasNoError());
+  const auto& file_list = file_list_or_error.value();
+  auto it = find(file_list.begin(), file_list.end(), tmp_file.file_path());
+  EXPECT_TRUE(it != file_list.end())
+      << "File " << tmp_file.file_path() << "was not found in the list: "
```
Comment (dpallotti) [inline src/OrbitBase/FileTest.cpp:314]:
```suggestion
      << "File " << tmp_file.file_path() << " was not found in the list: "
```

File: src/OrbitBase/FileTest.cpp [hunk 2]
```diff
@@ -267,4 +269,52 @@ TEST(File, ReadStructureAtOffset) {
...
+  EXPECT_TRUE(exists_or_error.value());
+
+  exists_or_error = FileExists(tmp_file.file_path());
+  ASSERT_THAT(exists_or_error, HasNoError());
+  EXPECT_FALSE(exists_or_error.value());
+
+  remove(new_path.string().c_str());
+}
+
+TEST(File, ListFilesInDirectory) {
```
Comment (dpallotti) [inline src/OrbitBase/FileTest.cpp:305]: Some vertical space in this test wouldn't hurt.

**Review by dpallotti — APPROVED** [review]

File: src/OrbitBase/FileTest.cpp (hunk 1 above)
Comment (dimitry-) [reply src/OrbitBase/FileTest.cpp]: done.

File: src/OrbitBase/FileTest.cpp (hunk 2 above)
Comment (dimitry-) [reply src/OrbitBase/FileTest.cpp:308]: done.

File: src/OrbitBase/TemporaryFile.cpp [hunk 3]
```diff
@@ -44,6 +44,9 @@ ErrorMessageOr<void> TemporaryFile::Init() {
         absl::StrFormat("Unable to create a temporary file: %s", SafeStrerror(errnum))};
   }
 
+  // get rid of '\0' at the end
+  file_path = std::filesystem::path(file_path.string().c_str());
```
Comment (danielfenner) [inline src/OrbitBase/TemporaryFile.cpp:48]: `file_path `is a string already. Perhaps make a copy of the `file_path `above and append the` \0` to that copy.

**Review by danielfenner — APPROVED** [review]
lgtm

File: src/OrbitBase/TemporaryFile.cpp (hunk 3 above)
Comment (dimitry-) [reply src/OrbitBase/TemporaryFile.cpp:48]: but then we will need to copy it over.. _mktemp_s modifies the string. Fixed by replacing fs::path with std::string() here.

**Pull request merged**